### PR TITLE
100 UI admin site maintenance notifications

### LINKED ISSUE #100: UI: Admin Site Maintenance Notifications
# 💡 Summary #

The Admin site maintenance tool should allow administrators to manage site maintenance messages. And create a method to issue a site management page. These messages will be displayed to users during scheduled maintenance on the login page  and/or when there are critical update a maintenance page will display.

## Motivation and context ##
This is valuable to streamline the maintenance processes for Crossfeed. 

## Implementation notes ##

- Create/Edit/Delete Messages: Administrators can create, edit, and delete maintenance messages.
- Scheduled Messages: Ability to schedule messages for specific dates and times.
-  Display Messages: Messages are prominently displayed to users during maintenance periods.
-  Notification System: Optionally, provide a notification system for users to be informed about upcoming maintenance.

No active or inactive notifications screen
image

Active notification and inactive notifications
image

Add New notification card and validation
image
image

Update notification pop up and validation
image
image

## Acceptance criteria ##

- [ ] Document all end-point changes
- [ ] Demonstrate a Response Design
- [ ] Include Screenshots Of Updated Views
- [ ] Commit to Staging
- [ ] Create Functional Tests

### PR DESCRIPTION
**Pull request by ameliav**
#  #

## 🗣 Description ##
The Admin site maintenance tool should allow administrators to manage site maintenance messages. And create a method to issue a site management page. These messages will be displayed to users during scheduled maintenance on the login page  and/or when there are critical update a maintenance page will display.

## 💭 Motivation and context ##
This is valuable to streamline the maintenance processes for Crossfeed. 

## 🧪 Testing ##
- Create/Edit/Delete Messages: Administrators can create, edit, and delete maintenance messages.
- Scheduled Messages: Ability to schedule messages for specific dates and times.
-  Display Messages: Messages are prominently displayed to users during maintenance periods.
-  Notification System: Optionally, provide a notification system for users to be informed about upcoming maintenance.

## 📷 Screenshots ##

No active or inactive notifications screen
image

Active notification and inactive notifications
image

Add New notification card and validation
image
image

Update notification pop up and validation
image
image

Login Screen with an Active notification
Screenshot 2024-05-10 at 1 21 11 PM

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [ ] Tests have been added and/or modified to cover the changes in this PR.
- [ ] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Create a release.

### CODE REVIEW

**Comment by ameliav** [comment]
@schmelz21 According to @rapidray12 any newly created endpoints are blocked until a particular team unblocks them. In order to not mess up staging, one possibility would be to revert the new /admin-tools endpoint to the use the existing /scans one until /admin-tools is available for use. What do you think?

**Comment by ameliav** [comment]
Blocked: implementation of stopping logins is blocked

**Review by schmelz21 — APPROVED** [review]
LGTM

File: backend/src/models/notification.ts
Comment (JCantu248) [inline backend/src/models/notification.ts:1]: Fix linter errors

File: backend/src/models/connection.ts [hunk 1]
```diff
@@ -2,6 +2,9 @@ import { createConnection, Connection } from 'typeorm';
 import {
   // Models for the Crossfeed database
   ApiKey,
+  SavedSearch,
```
Comment (JCantu248) [inline backend/src/models/connection.ts]: SavedSearch is a duplicate import. It is also being imported at line 19.

File: backend/src/models/connection.ts [hunk 2]
```diff
@@ -2,6 +2,9 @@ import { createConnection, Connection } from 'typeorm';
...
+  Notification,
+  OrganizationTag,
```
Comment (JCantu248) [inline backend/src/models/connection.ts]: OrganizationTag is a duplicate import. It is also being imported at line 14.

File: backend/src/models/connection.ts (hunk 1 above)
Comment (JCantu248) [reply backend/src/models/connection.ts]: I went ahead and resolved the duplicate issue myself and rebuilt. The change can be viewed on commit: c3af8b7375a248e9be731b1a20f00e01d9e43b21

File: backend/src/models/connection.ts (hunk 2 above)
Comment (JCantu248) [reply backend/src/models/connection.ts]: I went ahead and resolved the duplicate issue myself and rebuilt. The change can be viewed on commit: https://github.com/cisagov/XFD/commit/c3af8b7375a248e9be731b1a20f00e01d9e43b21

File: backend/src/models/notification.ts
Comment (JCantu248) [reply backend/src/models/notification.ts:1]: I corrected the linter errors myself in commit: cfb4c67a77e09b38c12ecd0d783b7de3b021fdca

**Review by JCantu248 — APPROVED** [review]
I fixed the duplicate imports error and the linter errors myself. Other than those small things, LGTM.

**Review by schmelz21 — APPROVED** [review]
end-to-end UAT passed.

**Comment by schmelz21** [comment]
UAT results
Screenshot 2024-05-14 at 6 19 19 PM
Screenshot 2024-05-14 at 6 14 52 PM

**Pull request merged**